web/src/compose_validate.ts

```diff
@@ -795,3 +795,92 @@ export function convert_mentions_to_silent_in_direct_messages(
     const silent_mention_text = people.get_mention_syntax(full_name, user_id, true);
     return silent_mention_text;
 }
+
+export const SPLIT_DELIMITER = "\n\n\n";
+
+let split_messages_enabled = false;
+
+export function is_split_messages_enabled(): boolean {
+    return split_messages_enabled;
+}
+
+/**
+ * sets the state of the option to the value passed, and
+ * also in the localstorage (if supported)
+ * @param {boolean} enable Whether to enable the option
+ */
+export function set_split_messages_enabled(enable: boolean): void {
+    split_messages_enabled = enable;
+}
+
+function delimiter_index_outside_code_block(message_content: string): number {
+    let index = message_content.indexOf(SPLIT_DELIMITER);
+    while (index !== -1) {
+        if (
+            !compose_ui.position_inside_code_block(message_content, index + 1) &&
+            !compose_ui.position_inside_code_block(message_content, index + 2)
```

Comment:
Actually it is sufficient to just check one position (index+1). I've removed the other one. (It was a remnant of earlier implementation).
I've added a comment explaining the while loop - its because we need to keep searching further until we find a delimiter which occurs outside of a code block.